src/support/path-utils.ts

```diff
@@ -11,31 +11,33 @@
  * subject to an additional IP rights grant found at
  * http://polymer.github.io/PATENTS.txt
  */
-import {relative} from 'path';
+import {posix, sep as pathSeparator} from 'path';
+
+const dirnameRegex = process.platform === 'win32' ? /[^\\]+$/ : /[^\/]+$/;
 
 /**
  * Similar to `path.dirname()` except includes trailing slash and for a
  * path `/like/this/` will return `/like/this/` instead of `/like` since the
  * trailing slash indicates `this` is a folder name not a file name.
  * (`path.dirname('/like/this/')` returns `/like`.)
  */
-export const dirname = (path: string): string => path.replace(/[^\/]+$/, '');
+export const dirname = (path: string): string => path.replace(dirnameRegex, '');
 
-export const ensureLeadingDot = (path: string): string =>
+export const ensureLeadingDotInURL = (path: string): string =>
     (path.startsWith('../') || path.startsWith('./')) ? path : './' + path;
 
-export const ensureTrailingSlash = (path: string): string =>
-    path.endsWith('/') ? path : path + '/';
+export const ensureTrailingSlashInPath = (path: string): string =>
+    path.endsWith(pathSeparator) ? path : path + pathSeparator;
 
 export const forwardSlashesOnlyPlease = (path: string): string =>
     path.replace(/\\/g, '/');
 
-export const getBasePath = (href: string): string =>
-    href.replace(/[^\/]+$/, '');
+export const getBaseUrl = (href: string): string => href.replace(/[^\/]+$/, '');
 
-export const noLeadingSlash = (href: string): string => href.replace(/^\//, '');
+export const noLeadingSlashInURL = (href: string): string =>
+    href.replace(/^\//, '');
 
-export const relativePath = (from: string, to: string): string =>
-    ensureLeadingDot(relative(
-        getBasePath(forwardSlashesOnlyPlease(from)),
+export const relativePathToUrl = (from: string, to: string): string =>
+    ensureLeadingDotInURL(posix.relative(
```

Comment:
The fix here is to use the `posix.relative` function instead of the platform specific `relative` function. `forwardSlashesOnlyPlease` converts all OS paths to URL paths, so every function that operates on the result of that function needs to assume URLs and not OS paths. I think `posix.relative` works well enough :)

Reply:
I named this function `relativePathToUrl` because it takes OS path parameters and returns a URL